### PR TITLE
Add table class to docs

### PR DESCRIPTION
**Pull request by hanscg**
Added table class to docs since this were the part where I was the most confused about. 
I'm thinking of adding sample code for query later.

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -59,6 +59,46 @@ In case you want to explicitly pass the credentials from Python, use :py:class:`
     :members:
     :undoc-members:
 
+The ``Table`` class
+-------------------
+
+After instantiating client, set the table to be used by calling the :py:meth:`aiodynamo.client.Client.table` method.
```
Comment (dimaqq) [inline docs/usage.rst:65]: Somehow... the language could be better...

File: src/aiodynamo/client.py [hunk 2]
```diff
@@ -73,6 +73,9 @@ class Table:
     name: TableName
 
     async def exists(self) -> bool:
+        """
+        Checks that the table exist.
```
Comment (dimaqq) [inline src/aiodynamo/client.py:77]:
```suggestion
        Checks if the table exists.
```

Comment (dimaqq) [reply src/aiodynamo/client.py:77]: I think that Python docs use the imperative mood, for example, `os.path.exists(...)`:
>     exists(path)
>        Test whether a path exists.  Returns False for broken symbolic links

File: docs/usage.rst [hunk 3]
```diff
@@ -59,6 +59,46 @@ In case you want to explicitly pass the credentials from Python, use :py:class:`
...
+.. automethod:: aiodynamo.client.Table.delete_item
+
+Read items
+~~~~~~~~~~
+
+These methods can be used to return items from the table.
+
+.. automethod:: aiodynamo.client.Table.get_item
+
+|
```
Comment (dimaqq) [inline docs/usage.rst:95]: What does the bar do?

Comment (ojii) [reply docs/usage.rst:95]: it inserts empty space in the output to make it less cluttered.


in sphinx: 

```
foo
|
bar
```

is similar to html:

```
<p>foo</p>
<p>bar</p>
```

whereas sphinx:

```
foo

bar
```

would be (irregardless of how many newlines between foo and bar) 

```
foo<br>bar
```

**Comment by dimaqq** [comment]
poke

**Comment by dimaqq** [comment]
@ojii I guess we have to pick this up...

**Comment by ojii** [comment]
> @ojii I guess we have to pick this up...

I know, I just have some doubts about the usefulness of documentation that says that `delete_table` "Deletes the table". And I especially don't think those docstrings are useful, so moving the explanation (especially of the arguments to the methods) should be in the docs imho, not the code.

**Comment by hanscg** [comment]
I'm sorry for leaving this for so long. 
I'll try to fix it according to your reviews next Monday.

File: docs/usage.rst [hunk 4]
```diff
@@ -59,6 +59,46 @@ In case you want to explicitly pass the credentials from Python, use :py:class:`
     :members:
     :undoc-members:
 
+The ``Table`` class
+-------------------
+
```
Comment (ojii) [inline docs/usage.rst:64]: please add a note that the `Table` class should not be instantiated directly, but always via Client.table.

File: docs/usage.rst [hunk 5]
```diff
@@ -59,6 +59,46 @@ In case you want to explicitly pass the credentials from Python, use :py:class:`
...
+After instantiating client, set the table to be used by calling the :py:meth:`aiodynamo.client.Client.table` method.
+The method takes the table name string as its only argument.
+
+.. automethod:: aiodynamo.client.Client.table
+
+Modify table
+~~~~~~~~~~~~
+
+These methods can be used to modify the table.
```
Comment (ojii) [inline docs/usage.rst:73]: this says the same as the "Modify table" header, so it's not needed

**Review by ojii — CHANGES_REQUESTED** [review]
maybe instead of the docstrings (which more or less say what the method name says), linking to the DynamoDB API docs (in the docs, not the code) might be more useful? In most cases, things map 1-to-1, so only the differences should be explained in detail in the docs. Off the top of my head, those are the auto-depagination in query/scan and the wait_for... in create/delete.

Also, a lot of datastructures used in arguments are not documented, they should at least get an autodoc.

**Comment by ojii** [comment]
thank you, merged.

**Pull request closed without merging**